c/tskit/trees.c

```diff
@@ -9256,6 +9256,7 @@ pair_coalescence_quantiles(tsk_size_t input_dim, const double *weight,
     j = 0;
     coalesced = 0.0;
     timepoint = -INFINITY;
+    /* TODO: may be more efficient to use a binary search */
```

Comment:
Just to note we have [tsk_search_sorted](https://github.com/tskit-dev/tskit/blob/46578bdc28211230ad7815b116564e57c2ba88e7/c/tskit/core.c#L826) if you want to try it out